lib/good_job/job.rb

```diff
@@ -209,10 +209,12 @@ def perform
 
       if rescued_error && GoodJob.reperform_jobs_on_standard_error
         save!
+      elsif rescued_error && GoodJob.preserve_job_records == :on_error
+        update!(finished_at: Time.current)
```

Comment:
I think this needs to be testing against `self.error`(any error that occurred during the job), rather than just `rescued_error` (errors that bubble up to the GoodJob backend).... or I think we should rename it specifically to be "on_backend_error"/"on_unhandled_error" (naming is hard).

Assuming that `self.error` matches with your needs for this, I think the entire block might be clearer with further rewriting (apologies it was messy in the first place). Something like:

```ruby
if rescued_error && GoodJob.reperform_jobs_on_standard_error
  save!
elsif GoodJob.preserve_job_records == true || (error && GoodJob.preserve_job_records == :on_error)
  self.finished_at = Time.current
  save!
else
  destroy!
end
```

Reply:
I actually wanted saving only on "rescued" aka "unhandled" error, so I renamed the config to be `:on_unhandled_error`, as otherwise it would create a bit of noise to have multiple records of  the same job/error.
It would be not clear if those entries are coming from the same AJ job or different ones.
This is matching the behavior of other AJ queue adapters.

I think storing on every error could also be an option (maybe another config value?) with some nice web UI grouping, so it's clear that given group of job exception is coming from the same AJ job.

Reply:
That is great. That makes me think (as a separate PR) that we should rename the configuration `GoodJob. reperform_jobs_on_standard_error` as `GoodJob.reperform_jobs_on_unhandled_error`. That configuration name was always unwieldy.

Reply:
@bensheldon Can I prepare a PR for renaming reperform_jobs_on_standard_error to reperform_jobs_on_unhandled_error with deprecation warning on old usage?

Reply:
@morgoth that would be great! 

Question: what are your thoughts about an even briefer `retry_on_unhandled_error`?